gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -437,6 +493,68 @@ public static ZoneFilter notEquals(ZoneField field, String value) {
   }
 
   /**
+   * Class for filtering operation lists.
+   */
+  class OperationFilter extends ListFilter {
+
+    private static final long serialVersionUID = -3202249202748346427L;
+
+    OperationFilter(OperationField field, ComparisonOperator operator, Object value) {
+      super(field.selector(), operator, value);
+    }
+
+    /**
+     * Returns an equality filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2">RE2</a>
+     */
+    public static OperationFilter equals(OperationField field, String value) {
+      return new OperationFilter(checkNotNull(field), ComparisonOperator.EQ, checkNotNull(value));
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2">RE2</a>
+     */
+    public static OperationFilter notEquals(OperationField field, String value) {
+      return new OperationFilter(checkNotNull(field), ComparisonOperator.NE, checkNotNull(value));
+    }
+
+    /**
+     * Returns an equality filter for the given field and long value.
+     */
+    public static OperationFilter equals(OperationField field, long value) {
+      return new OperationFilter(checkNotNull(field), ComparisonOperator.EQ, value);
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and long value.
+     */
+    public static OperationFilter notEquals(OperationField field, long value) {
+      return new OperationFilter(checkNotNull(field), ComparisonOperator.NE, value);
+    }
+
+    /**
+     * Returns an equality filter for the given field and integer value.
+     */
+    public static OperationFilter equals(OperationField field, int value) {
```

Comment:
because there is no enforcement (one could use the `int` version for a `long` property and vise versa) I think having only the `long` version is preferred.